### ISSUE TITLE
Can't remap to a sequence of keys that includes an i letter on the after part

### ISSUE THREAD

**Issue by danielo515** [VSCodeVim/Vim#4530]
**Describe the bug**
Hello.
I'm trying to remap a leader+b combination to a sequence of keys. The problem is that, some of those keys include commands that are context aware and that depend on the previous key. Seems that the plugin is not able to "remember" the previous letter and just treats each one in isolation, so if you after sequence includes an `i` it will put it in insert mode, no matter you previously have an `y`

**To Reproduce**
You can try this remap

```
"before": ["leader","b"], 
"after": ["m","b","`","a", "y", "i", "w", "`", "b", "v", "i", "w", "p", "`", "a", "v", "i", "w", "p", "`", "b" ]
```
**Expected behavior**
I want it to execute the keys as a sequence, not each key individually

**Screenshots**

instead what you get is yiw... etc because the first a

**Environment (please complete the following information):**

Extension version: 1.12.4
VS Code version: Code 1.41.1 (26076a4de974ead31f97692a0d32f90d735645c0, 2019-12-18T14:57:51.166Z)
OS version: Darwin x64 19.2.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz (12 x 2900)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|32.00GB (1.40GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Additional context**

**Comment by danielo515** [comment]
Funnily, if I change the order it works. This sequence works fine:

```
      "after": ["m","b","`","a","y","i","w","`","b","v","i","w","p","`","a","v","i","w","p","`","b"]
```

**Comment by berknam** [comment]
Both sequences you gave are the same. This works fine if you already have the 'a' mark set. If the 'a' mark is not set it will give the error 'E20: Mark not set' and then it should stop the remapping but currently it doesn't stop so it continues with 'a' being AppendAtCursor and then appends the rest of the keys to the buffer.
This will be fixed on PR #4735.
